src/control/server/harness.go

```diff
@@ -27,11 +33,52 @@ const (
 	rankStartTimeout = 2 * rankReqTimeout
 )
 
+// Engine defines an interface to be implemented by engine instances.
+//
+// NB: This interface is way too big right now; need to refactor in order
+// to limit scope.
+type Engine interface {
+	// These are definitely wrong... They indicate that too much internal
```

Comment:
rather than doing a refactor and leaving comments to the effect that things are "wrong", I think it's probably a more constructive approach to fix what you think needs fixing within your refactor

Reply:
I have a followup that builds on this. This PR is limited in scope to make it easy to review. Doesn't change any functionality, just introduces the interface to make followup work easier. Testing the harness is way more complicated than it should be because of all of these couplings across layers.

Reply:
One of the followups: #5911